test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/SerilogLoggerProviderTests.cs

```diff
@@ -105,31 +101,29 @@ public void DynamicLoggingLevel()
             var testSink = new Mock<ILogEventSink>(MockBehavior.Strict);
             testSink.Setup(m => m.Emit(It.IsAny<LogEvent>()));
 
-            using (var provider = new TestWebAppSerilogLoggerProvider(testSink.Object, configReader.Object))
-            {
-                var logger = provider.CreateLogger("TestLogger");
+            var provider = new TestWebAppSerilogLoggerProvider(testSink.Object);
+            var logger = provider.ConfigureLogger(configReader.Object);
 
-                logger.LogDebug("Test1");
-                testSink.Verify(m => m.Emit(It.IsAny<LogEvent>()), Times.Never);
+            logger.Debug("Test1");
+            testSink.Verify(m => m.Emit(It.IsAny<LogEvent>()), Times.Never);
 
-                configBuilder.SetFileLoggingLevel(LogLevel.Debug);
-                currentConfig = configBuilder.Build();
+            configBuilder.SetFileLoggingLevel(LogLevel.Debug);
+            currentConfig = configBuilder.Build();
 
-                configReader.Raise(m => m.OnConfigurationChanged += (sender, e) => { }, null, currentConfig);
+            configReader.Raise(m => m.OnConfigurationChanged += (sender, e) => { }, null, currentConfig);
 
-                // Logging for this level should be enabled now
-                logger.LogDebug("Test1");
-                testSink.Verify(m => m.Emit(It.IsAny<LogEvent>()), Times.Once);
-            }
+            // Logging for this level should be enabled now
+            logger.Debug("Test1");
+            testSink.Verify(m => m.Emit(It.IsAny<LogEvent>()), Times.Once);
         }
 
         // Checks that the .net log level to serilog level mappings are doing what we expect
         [Fact]
         public void LevelMapping()
```

Comment:
Instead of the loops in this test, couldn't you turn it into a `Theory` and generate the data for each `i,j` test case?